adapters/foo.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var Thing = require('../thing.js').Thing;
+
+class FooThing extends Thing {
+    constructor(adapter, name) {
+        let properties = {
+            'temperature': {
+                'type': 'integer',
+                'unit': 'celsius',
+                'description': 'An ambient temperature sensor',
+                'value': 23,
+            },
+            'humidity': {
+                'type': 'integer',
+                'unit': 'percent',
+                'value': 35,
+            },
+            'led': {
+                'type': 'boolean',
+                'description': 'A red LED',
+                'value': false,
+            },
+        };
+        super(adapter, 'FooThing', name, properties, [], []);
+    }
+
+    setPropertyValue(name, value) {
+        // This function should propogate the value to the hardware
+        // Calling super.setPropertyValue will update the value cache
+        // and send a 'value-changed' event to any listeners.
+        super.setPropertyValue(name, value);
+    }
+}
+
+class FooAdapter {
```

Comment:
These two classes should probably be in separate files. Also definitions of types of Thing should really be adapter (protocol) agnostic, with some kind of binding to each underlying protocol. So eventually thing definitions like this should probably live somewhere else outside of the adapters folder.